internal/pkg/dl/policies_integration_test.go

```diff
@@ -42,45 +41,111 @@ func createRandomPolicy(id string, revisionIdx int) model.Policy {
-		if err != nil {
-			return model.Policy{}, err
+func storeRandomPolicies(ctx context.Context, bulker bulk.Bulk, index string, count, maxRev int) error {
+	ops := make([]bulk.MultiOp, 0, count*maxRev)
+	for range count {
+		id := uuid.Must(uuid.NewV4()).String()
+		for i := 1; i <= maxRev; i++ {
+			rec := createRandomPolicy(id, i)
+			p, err := json.Marshal(&rec)
+			if err != nil {
+				return err
+			}
+			ops = append(ops, bulk.MultiOp{
+				Index: index,
+				Body:  p,
+			})
 		}
 	}
-	return rec, nil
+	_, err := bulker.MIndex(ctx, ops, bulk.WithRefresh())
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
+//nolint:dupl // test duplication
 func TestQueryLatestPolicies(t *testing.T) {
 	ctx := testlog.SetLogger(t).WithContext(t.Context())
 
-	index, bulker := ftesting.SetupCleanIndex(ctx, t, FleetPolicies)
+	index, bulker := ftesting.SetupCleanIndex(ctx, t, FleetPolicies, bulk.WithFlushThresholdCount(1))
+
+	err := storeRandomPolicies(ctx, bulker, index, 4, 4)
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(time.Second * 2) // FIXME ES does not refresh instantly?
+
+	policies, err := QueryLatestPolicies(ctx, bulker, WithIndexName(index))
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, policy := range policies {
+		if policy.RevisionIdx != 4 {
+			t.Errorf("Expected to find revision_idx 3 for policy %s, found %d", policy.PolicyID, policy.RevisionIdx)
+		}
+	}
+	if len(policies) != 4 {
+		t.Errorf("Expected 4 policies, got %d", len(policies))
+	}
+}
+
+// TestQueryLatestPolicies400k tests to see if  to see if the latest revision is correctly selected when a lot the revision count is very large
+//
+//nolint:dupl // test duplication
+func TestQueryLatestPolicies400k(t *testing.T) {
+	ctx := testlog.SetLogger(t).WithContext(t.Context())
+
+	index, bulker := ftesting.SetupCleanIndex(ctx, t, FleetPolicies, bulk.WithFlushThresholdCount(1))
 
-	recs := map[string]model.Policy{}
-	for i := 0; i < 0; i++ {
-		rec, err := storeRandomPolicy(ctx, bulker, index)
-		if err != nil {
-			t.Fatal(err)
+	err := storeRandomPolicies(ctx, bulker, index, 4, 100000)
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(time.Second * 2) // FIXME ES does not refresh instantly?
+
+	policies, err := QueryLatestPolicies(ctx, bulker, WithIndexName(index))
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, policy := range policies {
+		if policy.RevisionIdx != 100000 {
+			t.Errorf("Expected to find revision_idx 100000 for policy %s, found %d", policy.PolicyID, policy.RevisionIdx)
 		}
-		recs[rec.PolicyID] = rec
 	}
 
+	if len(policies) != 4 {
+		t.Errorf("Expected 4 policies, got %d", len(policies))
+	}
+}
+
+// TesyQueryLatestPolices11kUnique tests behaviour when 11k unique policies are used, there is a 10k size specification in the aggregation.
+//
+//nolint:dupl // test duplication
+func TestQueryLatestPolicies11kUnique(t *testing.T) {
+	t.Skip("Re-enable after policy load issues have been sorted: https://github.com/elastic/fleet-server/issues/3254")
```

Comment:
@blakerouse added the test for number of policies; currently it fails dues to the aggregation size (as expected).

I'm not sure in terms of priority where resolving the policy load issues is for us

Reply:
I think the case that there are 10,000 unique active policies is rare. I think its more likely that some automation creates and deletes lots of policies. If I remember correctly deleting a policy doesn't actually remove it from the index, it just sets it to `active: false`.

I wonder if it would be possible to remove all policies with `active: false` first before ES does the aggregation, allowing the size limit to be based on the number of active policies. If that is not possible, we might want to see about having some cleaner job that removes `active: false` policies after a period of time.

Reply:
Actually, policies do not have an active attribute in the ES template: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/template-resources/src/main/resources/fleet-policies.json

Reply:
I'm pretty sure that the `data.enabled` attribute in the template controls if the policy data is indexed

Reply:
I just checked with a cloud deployment; when a policy is deleted Kibana will delete all revisions of the policy

Reply:
Okay. Then this issue should only present itself in the case that user has 10,000+ active policies. That seems extremely unlikely.